core/src/test/java/org/infinispan/xsite/CacheOperationsTest.java

```diff
@@ -228,6 +230,19 @@ private void testFunctional(String site) {
       for (Object key : keys) {
          assertEquals(null, backup.get(key));
       }
+
+      if (transactional) {
+         TransactionManager tm = cache(site, 0).getAdvancedCache().getTransactionManager();
+         tm.begin();
+         rw.eval(keys[0], "v4", MarshallableFunctions.setValueReturnPrevOrNull()).join();
```

Comment:
Why don't you execute the operation on all keys in the set? (the test is supposed to test several combinations).

Reply:
because it wouldn't reproduce the NPE.

The NPE is caused by an `evalMany()` with a read-only function in in-existent keys. This will create an `CacheEntry` with `null` value in the `InvocationContext`.

The `eval(keys[0])` is to make sure the transaction is committed in the backup site with the read-only `evalMany()`.

Reply:
Do we have to miss *all* keys from the set to hit the NPE?

Reply:
no. but if I don't miss all key, I'll have to create a new function :) without any benefit for the test